metadata-ingestion/src/datahub/ingestion/source/superset.py

```diff
@@ -178,7 +182,9 @@ def __init__(self, ctx: PipelineContext, config: SupersetConfig):
         super().__init__(config, ctx)
         self.config = config
         self.report = StaleEntityRemovalSourceReport()
+        self.login()
```

Comment:
would be cleaner to make this `self.session = self.login()`

also the domain_registry logic should not be in the login method